atc/api/infoserver/info.go

```diff
@@ -15,6 +15,15 @@ func (s *Server) Info(w http.ResponseWriter, r *http.Request) {
 		WorkerVersion: s.workerVersion,
 		ExternalURL:   s.externalURL,
 		ClusterName:   s.clusterName,
+		FeatureFlags: map[string]bool{
```

Comment:
Very sensible approach putting it in the `info` response, but I'll propose the idea that first came to my mind. What if instead we interpolate the (relevant) flags into the elm flags directly when we interpolate `index.html`:
https://github.com/concourse/concourse/blob/49b742951de35f289a5d752f6aa17a6d211f0a6a/web/public/index.html#L73-L79

If we did it this way, we wouldn't need to wait for a full round trip to the ATC to make decisions. This isn't such a big deal here (the causality buttons may take a split second to load if you load the resource page directly), but waiting for a round trip isn't ideal for features like #5802.

Anyway, not a big deal, happy to keep it how to have it - the API-based approach could be used in places like `fly` as well, so has its own benefits.

Reply:
Yeah, the main problem with this approach is that the feature flags are not known at subpage init time. I wonder if we can have some sort of pre-init that blocks until `ClusterInfo` has been fetched. But that's an idea for another time

Reply:
I think rather than baking more into Elm with a pre-init or something, an approach like I suggested might be better. But anyway, for now, I think the current approach is A-OK